test/sql/function/string/test_string_split.test

```diff
@@ -0,0 +1,171 @@
+# name: test/sql/function/string/test_string_split.test
+# description: String split test
+# group: [string]
+
+statement ok
+PRAGMA enable_verification
+
+statement ok
+SELECT string_split(string_agg(range, 'DUCK '), ' '), mod(range, 100) xx FROM range(50000) GROUP BY xx
+
+statement ok
+SELECT string_split(string_agg(range, '🦆 '), ' '), mod(range, 100) xx FROM range(50000) GROUP BY xx
+
+statement ok
+SELECT string_split_regex(string_agg(range, 'DUCK '), '\s'), mod(range, 100) xx FROM range(50000) GROUP BY xx
+
+query T
+SELECT UNNEST(string_split('üüüüü', '◌̈')) 
+----
+üüüüü
+
+query T
+SELECT UNNEST(string_split('üüüüü', '◌'))
+----
+üüüüü
+
+query T
+SELECT UNNEST(string_split_regex('üüüüü', '◌̈')) 
+----
+üüüüü
+
+query T
+SELECT UNNEST(string_split_regex('üüüüü', '◌'))
+----
+üüüüü
+
+query T
+SELECT UNNEST(string_split(' 🦆🦆  🦆🦆', '  '))
+----
+ 🦆🦆
+🦆🦆
+
+query T
+SELECT UNNEST(string_split('a a a a a', ' '))
+----
+a
+a
+a
+a
+a
+
+query T
+SELECT UNNEST(string_split('🦆 🦆 🦆 🦆 🦆', ' '))
+----
+🦆
+🦆
+🦆
+🦆
+🦆
+
+query T
+SELECT UNNEST(string_split('🦆🐈🐈🦆🐈🐈🦆🐈🐈🦆🐈🐈🦆', '🐈🐈'))
+----
+🦆
+🦆
+🦆
+🦆
+🦆
+
+query T
+SELECT UNNEST(string_split('', 'delim'))
+----
+(empty)
+
+query T
+SELECT UNNEST(string_split('aaaaa', ''))
+----
+a
+a
+a
+a
+a
+
+query T
+SELECT UNNEST(string_split('🦆🦆🦆🦆🦆', ''))
+----
+🦆
+🦆
+🦆
+🦆
+🦆
+
+query T
+SELECT UNNEST(string_split('abab', 'b'))
+----
+a
+a
+(empty)
+
+query T
+SELECT UNNEST(string_split('🦆b🦆b', 'b'))
+----
+🦆
+🦆
+(empty)
+
+statement ok
+CREATE TABLE documents(s VARCHAR)
+
+statement ok
+INSERT INTO documents VALUES ('baabbaa'), ('aabbaab'), ('ababababa'), ('b🦆🦆bb🦆🦆'), ('🦆🦆bb🦆🦆b'), ('🦆b🦆b🦆b🦆b🦆')
+
+query T
+SELECT UNNEST(string_split(s, 'bb')) FROM documents WHERE 1
+----
+baa
+aa
+aa
+aab
+ababababa
+b🦆🦆
+🦆🦆
+🦆🦆
+🦆🦆b
+🦆b🦆b🦆b🦆b🦆
+
+query T
+SELECT UNNEST(string_split(s, 'bb')) FROM documents WHERE s LIKE 'b%'
+----
+baa
+aa
+b🦆🦆
+🦆🦆
+
+query T
+SELECT UNNEST(string_split_regex('a1a11a111a', '[0-9]+'))
+----
+a
+a
+a
+a
+
+query T
+SELECT UNNEST(string_split_regex('aaaaa', ''))
+----
+a
+a
+a
+a
+a
+
+
+query T
+SELECT UNNEST(string_split_regex('a a  a   a', '\s'))
+----
+a
+a
+a
+a
+
+query T
+SELECT UNNEST(string_split('aaaaa', NULL))
+----
+aaaaa
+
+#query T
+#SELECT UNNEST(string_split(NULL))
+#----
+#NULL
+#
+## TODO: full circle test with concat
```

Comment:
what's missing to make this possible?

Reply:
Also newline at end of file, code factor should complain about this

Reply:
Nothing, actually. Will write a test now